src/app/modules/common/services.js

```diff
@@ -535,6 +555,65 @@ angular.module('BloomLibraryApp.services', ['restangular'])
                 rBook.customPUT(putParams);
             };
 
+        this.relateBooksById = function() {
+            //Context of arguments is lost going into IIFE
+            var tArgs = arguments;
+
+            //Since, it might be possible for the user to click a second book
+            //faster than the server can process the request, I'm using an IIFE
+            (function() {
+                var args = tArgs;
+                var relatedBooks = [];
+
+                var baseRelatedBooks = restangular.withConfig(authService.config()).all('classes/relatedBooks');
+
+                var i = 0;
+
+                function handleOneId() {
```

Comment:
A comment should be added here explaining that a loop is being handled by recursion so that each iteration can use "then" to wait for results from parse.com